integration_tests/e2e/manageContactsRestrictions.cy.ts

```diff
@@ -0,0 +1,165 @@
+import SearchPrisonerPage from '../pages/searchPrisoner'
+import Page from '../pages/page'
+import TestData from '../../server/routes/testutils/testData'
+import ListContactsPage from '../pages/listContacts'
+import ManageContactDetailsPage from '../pages/manageContactDetails'
+import EnterRestrictionPage from '../pages/enterRestrictionPage'
+
+context('Manage contacts restrictions', () => {
+  const { prisonerNumber } = TestData.prisoner()
+  const contact = TestData.contact()
+  const prisonerContactId = 987654
+  const restrictionId = 555333
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn', { roles: ['PRISON'] })
+    cy.signIn()
+    cy.visit('/contacts/manage/prisoner-search/start')
+    cy.task('stubGetGenders')
+    cy.task('stubTitlesReferenceData')
+    cy.task('stubComponentsMeta')
+    cy.task('stubPrisoners', {
+      results: {
+        totalPages: 1,
+        totalElements: 1,
+        content: [TestData.prisoner()],
+      },
+      prisonId: 'HEI',
+      term: prisonerNumber,
+    })
+    cy.task('stubPrisonerById', TestData.prisoner())
+    cy.task('stubContactList', 'A1234BC')
+
+    cy.task('stubGetContactById', contact)
+    cy.task('stubGetPrisonerContactRelationshipById', { id: 31, response: TestData.prisonerContactRelationship() })
+  })
+  it(`should render restrictions tab on manage contact details`, () => {
+    cy.task('stubGetPrisonerContactRestrictions', {
+      prisonerContactId: 31,
+      response: {
+        prisonerContactRestrictions: [TestData.getContactRestrictionDetails({ contactId: contact.id })],
```

Comment:
there is a `TestData.getPrisonerContactRestrictionDetails`.  They are pretty much the same but the id field will be different, might be why tests are falling.

Reply:
updated to use the helper method